src/main/java/uk/ac/ebi/ena/webin/cli/service/CompleteSampleService.java

```diff
@@ -0,0 +1,209 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+import uk.ac.ebi.ena.webin.cli.service.exception.ServiceMessage;
+import uk.ac.ebi.ena.webin.cli.service.exception.ServiceiException;
+import uk.ac.ebi.ena.webin.cli.service.handler.NotFoundErrorHandler;
+import uk.ac.ebi.ena.webin.cli.service.utils.HttpHeaderBuilder;
+import uk.ac.ebi.ena.webin.cli.validator.reference.Attribute;
+import uk.ac.ebi.ena.webin.cli.validator.reference.Sample;
+
+ /**
+    This class is used to get the complete sample i.e sampleXml + sampleResponceObject
+    For this it will call samples/{id} and cli/reference/sample/{id} APIs
+ */
+public class CompleteSampleService extends WebinService
+{
+    private static final Pattern bioSampleIdPattern = Pattern.compile("^SAM[E,N,D][A-Z]?\\d{6,}$");
+    
+    public static class 
+    Builder extends AbstractBuilder<CompleteSampleService>
+    {
+        @Override public CompleteSampleService
+        build()
+        {
+            return new CompleteSampleService( this );
+        }
+    };
+
+    public CompleteSampleService(AbstractBuilder<CompleteSampleService> builder )
+    {
+        super( builder );
+    }
+
+    public Sample getCompleteSample(String sampleValue ) {
+        Sample sample= getSampleFromAPI(sampleValue, getTest());
+        return sample;
+    }
+
+    /**
+     * This method will call cli/reference/sample/{id} and get SampleResponse to build Sample object
+     * @param sampleValue
+     * @param test
+     * @return Sample object
+     */
+    private Sample getSampleFromAPI(String sampleValue, boolean test) {
+        
+        // The "cli/reference/sample/{id}" API accepts sampleId (or) sampleAlias (or) biosampleId and responds SampleResponse. 
+        RestTemplate restTemplate = new RestTemplate();
+        restTemplate.setErrorHandler(new NotFoundErrorHandler(
+                ServiceMessage.SAMPLE_SERVICE_VALIDATION_ERROR.format(sampleValue),
+                ServiceMessage.SAMPLE_SERVICE_SYSTEM_ERROR.format(sampleValue)));
+
+        HttpHeaders headers = getAuthTokenHeader();
+        ResponseEntity response = executeHttpGetSampleRsp(restTemplate, "cli/reference/sample/{id}", headers, sampleValue, test);
+
+        SampleResponse sampleResponse = (SampleResponse) response.getBody();
+        if (sampleResponse == null || !sampleResponse.canBeReferenced) {
+            throw ServiceiException.userError(ServiceMessage.SAMPLE_SERVICE_VALIDATION_ERROR.format(sampleValue));

```

Comment:
Think about the exceptions and how we should preserve the current error handling/reporting in Webin-CLI when we move its code from there to here.